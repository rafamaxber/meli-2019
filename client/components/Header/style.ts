import styled from 'styled-components';

import * as variables from '../MainLayout/styleGlobal'

export const MainHeader = styled.div`
  padding: 10px 0;
  background-color: ${({ theme }) => theme.meliColor};
`

export const Logo = styled.a`
  display: inline-block;
  font-size: 0;
  vertical-align: middle;

  @media (max-width: 530px) {
    display: block;
    text-align: center;
    margin-bottom: 10px;
  }
`

export const Img = styled.img``;
