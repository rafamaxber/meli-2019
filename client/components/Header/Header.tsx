import Link from 'next/link'
import SearchBar from '../SearchBar/SearchBar'
import * as Common from '../Common/main.styled'
import * as Styled from './style'

const Header = () => (
  <Styled.MainHeader>
    <Common.Limit>
      <Logo />
      <SearchBar />
    </Common.Limit>
  </Styled.MainHeader>
)

const Logo = () => (
  <Link href="/">
    <Styled.Logo role="link">
      <Styled.Img src="/static/img/Logo_ML.png" alt="Mercado Libre" title="Mercado Libre" />
    </Styled.Logo>
  </Link>
)

export default Header
