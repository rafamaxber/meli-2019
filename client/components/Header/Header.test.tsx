import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';

describe('Header', () => {
  describe('smoke tests', () => {
    test('should render component with Header', () => {
      const component = render(<Header/>);

      expect(component).toMatchSnapshot();
    })
  });
})


