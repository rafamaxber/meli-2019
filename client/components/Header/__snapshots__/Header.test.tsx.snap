// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header smoke tests should render component with Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c2 {
  display: inline-block;
  max-width: 900px;
  vertical-align: middle;
  left: 26px;
  width: 90%;
  position: relative;
}

.c3 {
  height: 35px;
  padding: 10px;
  width: 95.5%;
  border: none;
  border-radius: undefined 0 0 undefined;
}

.c3::-webkit-input-placeholder {
  font-size: 18px;
  padding-top: 2px;
  position: relative;
}

.c3::-moz-placeholder {
  font-size: 18px;
  padding-top: 2px;
  position: relative;
}

.c3:-ms-input-placeholder {
  font-size: 18px;
  padding-top: 2px;
  position: relative;
}

.c3::placeholder {
  font-size: 18px;
  padding-top: 2px;
  position: relative;
}

.c3:focus {
  outline: none;
  box-shadow: -2px 1px 10px -5px;
}

.c4 {
  border: none;
  cursor: pointer;
  border-radius: 0 undefined undefined 0;
  padding: 10px;
  display: inline-block;
  vertical-align: top;
  font-size: 0;
  height: 35px;
  position: relative;
}

.c0 {
  padding: 10px 0;
}

.c1 {
  display: inline-block;
  font-size: 0;
  vertical-align: middle;
}

@media (max-width:939px) {
  .c3 {
    width: 85%;
  }
}

@media (max-width:530px) {
  .c1 {
    display: block;
    text-align: center;
    margin-bottom: 10px;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class=""
        >
          <a
            class="c1"
            role="link"
          >
            <img
              alt="Mercado Libre"
              class=""
              src="/static/img/Logo_ML.png"
              title="Mercado Libre"
            />
          </a>
          <div
            class="c2"
          >
            <form
              action="/items"
              class=""
              method="GET"
              role="search"
            >
              <input
                aria-autocomplete="list"
                aria-required="true"
                autocapitalize="off"
                autocorrect="off"
                class="c3"
                data-testid="search"
                maxlength="120"
                name="search"
                placeholder="Buscar Producto"
                required=""
                type="search"
              />
              <button
                aria-label="Buscar"
                class="c4"
                type="submit"
              >
                <img
                  src="/static/img/ic_Search.png"
                />
                <span
                  class=""
                  hidden=""
                >
                  Buscar
                </span>
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="style__MainHeader-sc-1dia5j1-0 kQfplN"
    >
      <div
        class="mainstyled__Limit-kgem6r-2 poVdl"
      >
        <a
          class="style__Logo-sc-1dia5j1-1 exRDuv"
          role="link"
        >
          <img
            alt="Mercado Libre"
            class="style__Img-sc-1dia5j1-2 eirKPw"
            src="/static/img/Logo_ML.png"
            title="Mercado Libre"
          />
        </a>
        <div
          class="style__SearchBar-sc-286i50-0 dVhlOI"
        >
          <form
            action="/items"
            class="style__Form-sc-286i50-1 bxazgb"
            method="GET"
            role="search"
          >
            <input
              aria-autocomplete="list"
              aria-required="true"
              autocapitalize="off"
              autocorrect="off"
              class="style__Input-sc-286i50-2 fogGJj"
              data-testid="search"
              maxlength="120"
              name="search"
              placeholder="Buscar Producto"
              required=""
              type="search"
            />
            <button
              aria-label="Buscar"
              class="style__Button-sc-286i50-3 kLGETb"
              type="submit"
            >
              <img
                src="/static/img/ic_Search.png"
              />
              <span
                class="style__Typography-sc-286i50-4 daqenF"
                hidden=""
              >
                Buscar
              </span>
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
